### PR TITLE
HOSTEDCP-2040: [CPO Refactor] Refactor HCCO

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
Refactor HCCO Component

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Review by jparrill — COMMENTED** [review]
The refactor looks amazing, I'm still reviewing it xD

File: control-plane-operator/controllers/hostedcontrolplane/testdata/hosted-cluster-config-operator/zz_fixture_TestControlPlaneComponents.yaml [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        key: hypershift.openshift.io/cluster
+        operator: Equal
+        value: hcp-namespace
+      volumes:
+      - name: kubeconfig
+        secret:
+          defaultMode: 416
+          secretName: hcco-kubeconfig
+      - configMap:
+          defaultMode: 420
```
Comment (csrwng) [inline control-plane-operator/controllers/hostedcontrolplane/testdata/hosted-cluster-config-operator/zz_fixture_TestControlPlaneComponents.yaml:180]: nit: we should be consistent with defaultMode (looks like we use 416 and 420 all over the place), in this deployment we have 2 configmaps with different defaultMode

File: control-plane-operator/controllers/hostedcontrolplane/v2/configoperator/podmonitor.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import (
+	component "github.com/openshift/hypershift/support/controlplane-component"
+	"github.com/openshift/hypershift/support/metrics"
+	"github.com/openshift/hypershift/support/util"
+
+	prometheusoperatorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+)
+
+func adaptPodMonitor(cpContext component.ControlPlaneContext, podMonitor *prometheusoperatorv1.PodMonitor) error {
+	podMonitor.Spec.NamespaceSelector = prometheusoperatorv1.NamespaceSelector{MatchNames: []string{cpContext.HCP.Namespace}}
```
Comment (csrwng) [inline control-plane-operator/controllers/hostedcontrolplane/v2/configoperator/podmonitor.go:12]: If we're going to do similar adapts for pod/service monitors, maybe it makes sense to have a common function to call

File: control-plane-operator/controllers/hostedcontrolplane/v2/configoperator/reconcile.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		v.VolumeSource.Secret.SecretName = manifests.HCCOKubeconfigSecret("").Name
+	})
+	util.UpdateVolume(rootCAVolumeName, deployment.Spec.Template.Spec.Volumes, func(v *corev1.Volume) {
+		v.VolumeSource.ConfigMap.Name = manifests.RootCAConfigMap("").Name
+	})
+	util.UpdateVolume(clusterSignerCAVolumeName, deployment.Spec.Template.Spec.Volumes, func(v *corev1.Volume) {
+		v.VolumeSource.ConfigMap.Name = manifests.KubeletClientCABundle("").Name
+	})
+
+	if isExternalInfraKv(hcp) {
```
Comment (csrwng) [inline control-plane-operator/controllers/hostedcontrolplane/v2/configoperator/reconcile.go:155]: nit: Kv in this context is not immediately clear... can we expand it to Kubevirt?

**Review by csrwng — COMMENTED** [review]
Some nits. I don't feel strongly about them, so not blockers

File: control-plane-operator/controllers/hostedcontrolplane/testdata/hosted-cluster-config-operator/zz_fixture_TestControlPlaneComponents.yaml (hunk 1 above)
Comment (enxebre) [reply control-plane-operator/controllers/hostedcontrolplane/testdata/hosted-cluster-config-operator/zz_fixture_TestControlPlaneComponents.yaml:180]: I think this should be possibly enforced by code. Not to block this PR.

Comment (muraee) [reply control-plane-operator/controllers/hostedcontrolplane/testdata/hosted-cluster-config-operator/zz_fixture_TestControlPlaneComponents.yaml:180]: makes sense, added a centralized function to enforce the same defaultMode.

File: control-plane-operator/controllers/hostedcontrolplane/v2/configoperator/podmonitor.go (hunk 2 above)
Comment (muraee) [reply control-plane-operator/controllers/hostedcontrolplane/v2/configoperator/podmonitor.go]: done

**Pull request merged**